src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -113,8 +113,17 @@ public ValueTask<IJSObjectReference> InvokeNewAsync(string identifier, Cancellat
         if (DefaultAsyncTimeout.HasValue)
         {
             using var cts = new CancellationTokenSource(DefaultAsyncTimeout.Value);
-            // We need to await here due to the using
-            return await InvokeAsync<TValue>(targetInstanceId, identifier, callType, cts.Token, args);
+            
+            try
+            {
+                // We need to await here due to the using
+                return await InvokeAsync<TValue>(targetInstanceId, identifier, callType, cts.Token, args);
+            }
+            catch (OperationCanceledException) when (cts.Token.IsCancellationRequested)
+            {
+                // This was cancelled due to our timeout, throw a more meaningful exception
+                throw new TimeoutException("A JavaScript interop call timed out. Consider increasing the timeout duration if the operation is expected to take longer.");
+            }
```

Comment:
I think we should do the exception check in ComponentBase.cs like Safia mentioned in  https://github.com/dotnet/aspnetcore/issues/21384#issuecomment-665819761

The current code won't work if you e.g. pass in your own CancellationToken.